sql/distinct.go

```diff
@@ -37,6 +37,9 @@ func (*planner) distinct(n *parser.Select, p planNode) planNode {
 		d.columnsInOrder = make([]bool, len(p.Columns()))
 	}
 	for _, p := range p.Ordering() {
+		if p == 0 {
```

Comment:
I'm open to hearing concrete proposals for improvement. Perhaps I'm misunderstanding your criticism, but I personally don't think getting rid of the layers of plans will make the code clearer. In fact, I think doing so will make the code more complex.